src/modules/previewpane/GcodePreviewHandler/GcodePreviewHandlerControl.cs

```diff
@@ -46,52 +46,50 @@ public override void DoPreview<T>(T dataSource)
             if (global::PowerToys.GPOWrapper.GPOWrapper.GetConfiguredGcodePreviewEnabledValue() == global::PowerToys.GPOWrapper.GpoRuleConfigured.Disabled)
             {
                 // GPO is disabling this utility. Show an error message instead.
-                InvokeOnControlThread(() =>
-                {
-                    _infoBarAdded = true;
-                    AddTextBoxControl(Properties.Resource.GpoDisabledErrorText);
-                    Resize += FormResized;
-                    base.DoPreview(dataSource);
-                });
+                _infoBarAdded = true;
+                AddTextBoxControl(Properties.Resource.GpoDisabledErrorText);
+                Resize += FormResized;
+                base.DoPreview(dataSource);
```

Comment:
These no longer need to be done in the control thread?

Reply:
correct. Control thread is the same thread basically, so doesn't harm.. I'll remove it

Reply:
removed